pkg/traceql/engine_metrics.go

```diff
@@ -1484,3 +1505,119 @@ func FloatizeAttribute(s Span, a Attribute) (float64, StaticType) {
 	}
 	return f, v.Type
 }
+
+// processTopK implements TopKBottomK topk method
+func processTopK(input SeriesSet, limit int) SeriesSet {
+	var valueLength int
+	for _, series := range input {
+		valueLength = len(series.Values)
+		break
+	}
+
+	result := make(SeriesSet)
+
+	// process each timestamp
+	for i := 0; i < valueLength; i++ {
+		// Min heap for top-k (smallest values at top for easy replacement)
+		h := &seriesHeap{}
+		heap.Init(h)
+
+		// process each series for this timestamp
+		for key, series := range input {
+			if i >= len(series.Values) {
+				continue
+			}
+
+			value := series.Values[i]
+			if math.IsNaN(value) {
+				continue // Skip NaN values
+			}
+
+			// If heap not full yet, add the value
+			if h.Len() < limit {
+				heap.Push(h, seriesValue{
+					key:   key,
+					value: value,
+				})
+				continue
+			}
+
+			// If new value is greater than smallest in heap, replace it
+			smallest := (*h)[0]
+			if value > smallest.value {
+				heap.Pop(h)
+				heap.Push(h, seriesValue{
+					key:   key,
+					value: value,
+				})
+			}
+		}
+
+		// we have iterated over all series for this timestamp
+		// empty the heap and record these series in result set
+		for h.Len() > 0 {
+			sv := heap.Pop(h).(seriesValue)
+			result[sv.key] = input[sv.key]
```

Comment:
This is adding the entire series to the result, instead of just the 1 timestamp value.  I think it should be something like `result[sv.key].Values[i] = input[sv.key].Values[i]`. The first time the series is added to the result it needs to be initialized with valueLength NaNs, and any still unset at the end are dropped further up the chain.